EssentialsDiscord/README.md

```diff
@@ -0,0 +1,442 @@
+response. Additionally, you can also delete the responses once you're done looking at them, so they don't clutter
+your chat.
+
+However, this is all configurable! In the `commands` section of the config, lies a ton of options to configure
+settings on a per-command basis. Below are explanations of what all the configuration options mean and how to use
+them.
+
+* `enabled`
+  * Default: `true`
+  * Description: `Whether or not the command should be enabled and therefore shown on discord. Note that you
+    must restart your minecraft server before this option takes effect.`
+* `hide-command`
+  * Default: `true`
+  * Description: `Whether other people should not be able to see what commands you execute. Setting to false
+    would allow people in the same channel as you to see exactly what command you execute. In the example below,
+    you can see how disabling this option shows a message of the user and the command they executed.`
+  * Example: ![Show Command](https://i.imgur.com/Q61iP4n.gif)
+* `allowed-roles`
+  * Description: `A list of user ids or role names/ids that are allowed to use the command. You can also use '*'
+    in order to allow everyone to use the command.`
+* `admin-roles`
+  * `A list of user ids or role names/ids that have extra features in the command. For example, in the list
+    command, admin-roles allows people to see vanished players.`
+
+---
+
+## Misc Permissions
+EssentialsXDiscord has a few miscellanies permissions that may be important to know about;
+
+* `essentials.discord.markdown` - Allows players to bypass the markdown filter, so that they can 
+bold/underline/italic/etc their minecraft chat messages for discord.
+* `essentials.discord.ping` - Allows players to bypass the ping filter, so that they can ping @everyone/@here
+from minecraft chat.
+
+---
+
+## Developer API
+EssentialsXDiscord has a pretty extensive API which allows any third party plugin to build
+their own integrations into it. Outside the specific examples below, you can also view
+javadocs for EssentialsXDiscord [here](https://jd-v2.essentialsx.net/EssentialsDiscord).
+
+### Sending Messages to Discord
+EssentialsXDiscord organizes the types of messages that can be sent along with their 
+destination on discord under the `message-types` section of the `config.yml`. The 
+EssentialsXDiscord API uses `message-types` to resolve the channel id you want to send your
+message to.
+
+#### Using a built-in message channel
+EssentialsXDiscord defines a few built in `message-types` which you may fit your use case
+already (such as sending a message to the mc->discord chat relay channel). The list of 
+built-in message types can be found at [`DiscordMessageEvent.MessageType.DefaultTypes`](https://github.com/EssentialsX/Essentials/blob/2.x/EssentialsDiscord/src/main/java/net/essentialsx/api/v2/events/discord/DiscordMessageEvent.java#L195-L203).
+
+Here is an example of what sending a message to the built-in chat channel would look like:
```

Comment:
I'm not sure how I feel about firing an event to trigger a message - I'd prefer making an explicit method call, and then recommending the event to *listen* to messages that are being sent

Reply:
I can provide a helper method if we *want* but due this approach just cuts out the middleman and is cleaner imo.